tasks/RTPTask.cpp

```diff
@@ -0,0 +1,146 @@
+    fetchUnsignedInt(stats, "recv_fir_count", &m_rtp_statistics.recv_fir_count);
+    fetchUnsignedInt(stats, "sent_nack_count", &m_rtp_statistics.sent_nack_count);
+    fetchUnsignedInt(stats, "recv_nack_count", &m_rtp_statistics.recv_nack_count);
+    fetchUnsignedInt(stats, "reck_packet_rate", &m_rtp_statistics.recv_packet_rate);
+    fetchBoolean(stats, "have_sr", &m_rtp_statistics.have_sr);
+    fetchUnsignedInt(stats, "in_ntptime", &m_rtp_statistics.in_ntptime);
+    fetchUnsignedInt(stats, "sr_rtptime", &m_rtp_statistics.sr_rtptime);
+    fetchUnsignedInt(stats, "sr_octet_count", &m_rtp_statistics.sr_octet_count);
+    fetchUnsignedInt(stats, "sr_packet_count", &m_rtp_statistics.sr_packet_count);
+    fetchBoolean(stats, "sent_rb", &m_rtp_statistics.sent_rb);
+    fetchUnsignedInt(stats,
+        "sent_rb_fractionlost",
+        &m_rtp_statistics.sent_rb_fractionlost);
+    fetchUnsignedInt(stats,
+        "sent_rb_packets_lost",
+        &m_rtp_statistics.sent_rb_packetslost);
+    fetchUnsignedInt(stats,
+        "sent_rb_exthighestseq",
+        &m_rtp_statistics.sent_rb_exthighestseq);
+    fetchUnsignedInt(stats, "sent_rb_jitter", &m_rtp_statistics.sent_rb_jitter);
+    fetchUnsignedInt(stats, "sent_rb_lsr", &m_rtp_statistics.sent_rb_lsr);
+    fetchUnsignedInt(stats, "sent_rb_dlsr", &m_rtp_statistics.sent_rb_dlsr);
+}
+
+void RTPTask::fetchBoolean(const GstStructure* structure,
+    const char* fieldname,
+    bool *boolean)
+{
+    gboolean gboolean_value;
+    if (gst_structure_get_boolean(structure, fieldname, &gboolean_value)) {
+        *boolean = gboolean_value == TRUE;
+    }
+    LOG_INFO_S << "Field: \"" << fieldname << "\" not found or not a boolean";
+}
+
+void RTPTask::fetchUnsignedInt(const GstStructure* structure,
+    const char* fieldname,
+    int *number)
+{
+    guint guint_value;
+    if (gst_structure_get_uint(structure, fieldname, &guint_value)) {
+        *number = static_cast<int>(guint_value);
+    }
+    LOG_INFO_S << "Field: \"" << fieldname << "\" not found or not a uint";
+}
+
+void RTPTask::fetchString(const GstStructure* structure,
+    const char* fieldname,
+    std::string *str_var)
+{
+    const gchar* gstr_value = gst_structure_get_string(structure, fieldname);
+    if (gstr_value != nullptr) {
+        *str_var = std::string(gstr_value);
+    }
+    LOG_INFO_S << "Field: \"" << fieldname << "\" not found or not a string";
+}
+
+bool RTPTask::configureHook()
+{
+    m_rtpbin = gst_bin_get_by_name(GST_BIN(m_pipeline), "rtpbin");
+    if (!m_rtpbin) {
+        throw std::runtime_error("cannot find element named rtpbin in pipeline");
+    }
+
+    if (!RTPTaskBase::configureHook())
+        return false;
+    return true;
+}
+
+bool RTPTask::startHook()
+{
+    if (!RTPTaskBase::startHook())
+        return false;
+    return true;
+}
+
+void RTPTask::updateHook()
+{
+    extractPipelineStatus();
+    _rtp_statistics.write(m_rtp_statistics);
```

Comment:
the following would be more likely to fit out coding standards
```suggestion
    RTPStreamStatistics stats = RTPStats(m_rtpbin); // this is the same of extractPipelineStatus
    _rtp_statistics.write(stats);
```

The name `extractPipelineStatus` is not accurate because a pipeline is composed by multiple elements and what you are really doing is getting rtp stats from _just_ the rtpbin element